### ISSUE TITLE
Cannot restore NuGet packages with dotnet restore command

### ISSUE THREAD

**Issue by bodnarpolina** [dotnet/sdk#30027]
### Describe the bug
Running _dotnet restore_ leads to " warning : Unable to find a project to restore!"

### To Reproduce
There is a solution with 5 projects:
![image](https://user-images.githubusercontent.com/70271320/213477321-ebc70ac4-12a0-43a9-8b42-6e267f2575fe.png)
4 of these projects contains NuGet packages I'm trying to restore.
Here is what I get:

```
D:\TreatConcierge>dotnet restore Example.sln
Determining projects to restore...
C:\Program Files\dotnet\sdk\6.0.404\NuGet.targets(132,5): warning : Unable to find a project to restore! [D:\TreatConci
erge\Example.sln]
```

### Further technical details
.NET SDK (reflecting any global.json):
 Version:   6.0.404
 Commit:    be4f3ec411

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.404\

global.json file:
  Not found

Host:
  Version:      6.0.12
  Architecture: x64
  Commit:       02e45a41b7

.NET SDKs installed:
  5.0.415 [C:\Program Files\dotnet\sdk]
  6.0.404 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info

**Comment by baronfel** [comment]
Hi @bodnarpolina  - do you have instructions we can follow to replicate the project/solution layout you have here? Or maybe a zip of your local directory structure?

**Comment by bodnarpolina** [comment]
Hi, @baronfel! Here it is
[Example.zip](https://github.com/dotnet/sdk/files/10458395/Example.zip)

**Comment by baronfel** [comment]
Thanks! I took a look and it looks like the projects you're using in your example are not SDK-style projects. The `dotnet` CLI doesn't support building these kinds of projects - instead you will have to use the `msbuild.exe` shipped with Visual Studio to restore, build, etc these projects.

**Comment by bodnarpolina** [comment]
Thank you for your quick answer! But it is strange because on MSDN page : https://learn.microsoft.com/en-us/nuget/consume-packages/install-use-packages-nuget-cli I found this information: _For non-SDK-style projects that use PackageReference instead of packages.config for package references, use the [dotnet CLI](https://learn.microsoft.com/en-us/nuget/consume-packages/install-use-packages-dotnet-cli) instead._

By the way I tried MSbuild and got the same error. But then I tried this:

`MSBuild.exe C:\Users\pbodnar\Documents\treatconcierge\Example.sln -t:restore -p:Configuration=Release -p:Platform=x64`

and had no errors. I mean I manually set the Configuration and Platform.
